File: Exercise/Medium/11. Count Occurrences in Array/Code/Count Occurrences in Array.java
import java.util.Scanner;

public class Main {
    static void readVector(int[] numbers) {
        Scanner scanner = new Scanner(System.in);
        for (int i = 0; i < numbers.length; i++) {
            numbers[i] = scanner.nextInt();
        }
    }

    static void findNumber(int[] numbers, int num) {
        int counter = 0;
        for (int number : numbers) {
            if (number == num) {
                counter++;
            }
        }
        System.out.println(counter);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] numbers = new int[n];

        readVector(numbers);

        int num = scanner.nextInt();

        findNumber(numbers, num);
    }
}

File: Exercise/Medium/11. Count Occurrences in Array/Code/Count Occurrences in Array.cpp
#include <iostream>
#include <vector>
#include <sstream>
#include <string>

using namespace std;

void readVector(vector<int> & numbers) {
    string line;
    getline(cin, line);
    istringstream istr(line);

    int number;
    while(istr >> number) {
        numbers.push_back(number);
    }
}

void findNumber(vector<int> numbers) {
    int num;
    cin >> num;

    int counter = 0;
    while (!numbers.empty()) {
        if (numbers.back() == num) {
            counter++;
        }
        numbers.pop_back();
    }

    cout << counter << endl;
}

int main() {
    vector<int> numbers;

    readVector(numbers);
    findNumber(numbers);

    return 0;
}